### ISSUE TITLE
Cannot compile django 2.0.2 "AttributeError: 'ExpressionOutlineFunction' object has no attribute 'getParameters'"

### ISSUE THREAD

**Issue by pwhipp** [kayhayen/Nuitka#53]
nuitka: 0.5.28.1 (installed via pip install...)
linux: Ubuntu 16.04
python: 3.5.2

Install new django in virtualenv running python 3.5.2. (just django and nuitka installed)
Create python file `do_django.py` containing django-admin.py content:
```
#!/usr/bin/env python
from django.core import management

if __name__ == "__main__":
    management.execute_from_command_line()
```
```
(nuitka_django) ~/wk/cliosoft/related_projects/nuitka/django $ nuitka --recurse-all do_django.py 
Nuitka:WARNING:/home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/django/utils/autoreload.py:59: Cannot find 'pyinotify' in package 'django.utils' as absolute import.
Nuitka:WARNING:/home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/django/core/validators.py:502: Cannot find 'PIL' in package 'django.core' as absolute import.
Nuitka:WARNING:/home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/django/template/backends/jinja2.py:1: Cannot find 'jinja2' in package 'django.template.backends' as absolute import.
Nuitka:WARNING:/home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/django/forms/fields.py:611: Cannot find 'PIL' in package 'django.forms' as absolute import.
Nuitka:WARNING:/home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/django/db/models/fields/files.py:372: Cannot find 'PIL' in package 'django.db.models.fields' as absolute import.
Nuitka:WARNING:/home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/django/core/files/images.py:39: Cannot find 'PIL' in package 'django.core.files' as absolute import.
Problem with statement at /home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/django/db/models/aggregates.py:26:
Problem with statement at /home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/django/db/models/aggregates.py:26:
Problem with statement at /home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/django/db/models/aggregates.py:26:
Problem with statement at /home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/django/db/models/aggregates.py:24:
Problem with statement at /home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/django/db/models/aggregates.py:24:
Problem with statement at /home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/django/db/models/aggregates.py:14:
Problem with statement at /home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/django/db/models/aggregates.py:14:
Nuitka:INFO:Interrupted while working on '<Node 'COMPILED_PYTHON_MODULE' with {'name': 'aggregates', 'package': 'django.db.models', 'filename': '/home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/django/db/models/aggregates.py'}>'.
Traceback (most recent call last):
  File "/home/paul/.virtualenvs/nuitka_django/bin/nuitka", line 211, in <module>
    MainControl.main()
  File "/home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/nuitka/MainControl.py", line 758, in main
    filename = filename
  File "/home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/nuitka/MainControl.py", line 118, in createNodeTree
    Optimization.optimize()
  File "/home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/nuitka/optimizations/Optimization.py", line 468, in optimize
    finished = makeOptimizationPass(False)
  File "/home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/nuitka/optimizations/Optimization.py", line 382, in makeOptimizationPass
    changed = optimizeModule(current_module)
  File "/home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/nuitka/optimizations/Optimization.py", line 168, in optimizeModule
    changed = optimizeCompiledPythonModule(module)
  File "/home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/nuitka/optimizations/Optimization.py", line 92, in optimizeCompiledPythonModule
    module.computeModule()
  File "/home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/nuitka/nodes/ModuleNodes.py", line 439, in computeModule
    trace_collection = self.trace_collection
  File "/home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/nuitka/nodes/StatementNodes.py", line 165, in computeStatementsSequence
    trace_collection
  File "/home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/nuitka/nodes/FrameNodes.py", line 194, in computeStatementsSequence
    statement = statement
  File "/home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/nuitka/optimizations/TraceCollections.py", line 566, in onStatement
    statement.computeStatement(self)
  File "/home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/nuitka/nodes/TryNodes.py", line 128, in computeStatement
    trace_collection = trace_collection
  File "/home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/nuitka/nodes/StatementNodes.py", line 169, in computeStatementsSequence
    statement = statement
  File "/home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/nuitka/optimizations/TraceCollections.py", line 566, in onStatement
    statement.computeStatement(self)
  File "/home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/nuitka/nodes/AssignNodes.py", line 284, in computeStatement
    trace_collection.onExpression(self.getAssignSource())
  File "/home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/nuitka/optimizations/TraceCollections.py", line 535, in onExpression
    trace_collection = self
  File "/home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/nuitka/nodes/OutlineNodes.py", line 244, in computeExpressionRaw
    trace_collection = trace_collection
  File "/home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/nuitka/nodes/StatementNodes.py", line 169, in computeStatementsSequence
    statement = statement
  File "/home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/nuitka/optimizations/TraceCollections.py", line 566, in onStatement
    statement.computeStatement(self)
  File "/home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/nuitka/nodes/AssignNodes.py", line 284, in computeStatement
    trace_collection.onExpression(self.getAssignSource())
  File "/home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/nuitka/optimizations/TraceCollections.py", line 535, in onExpression
    trace_collection = self
  File "/home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/nuitka/nodes/ExpressionBases.py", line 820, in computeExpressionRaw
    expression = sub_expression
  File "/home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/nuitka/optimizations/TraceCollections.py", line 535, in onExpression
    trace_collection = self
  File "/home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/nuitka/nodes/FunctionNodes.py", line 881, in computeExpressionRaw
    trace_collection = trace_collection
  File "/home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/nuitka/nodes/StatementNodes.py", line 169, in computeStatementsSequence
    statement = statement
  File "/home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/nuitka/optimizations/TraceCollections.py", line 566, in onStatement
    statement.computeStatement(self)
  File "/home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/nuitka/nodes/TryNodes.py", line 128, in computeStatement
    trace_collection = trace_collection
  File "/home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/nuitka/nodes/StatementNodes.py", line 165, in computeStatementsSequence
    trace_collection
  File "/home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/nuitka/nodes/FrameNodes.py", line 194, in computeStatementsSequence
    statement = statement
  File "/home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/nuitka/optimizations/TraceCollections.py", line 566, in onStatement
    statement.computeStatement(self)
  File "/home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/nuitka/nodes/ReturnNodes.py", line 56, in computeStatement
    trace_collection.onExpression(self.getExpression())
  File "/home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/nuitka/optimizations/TraceCollections.py", line 535, in onExpression
    trace_collection = self
  File "/home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/nuitka/nodes/OutlineNodes.py", line 244, in computeExpressionRaw
    trace_collection = trace_collection
  File "/home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/nuitka/nodes/StatementNodes.py", line 169, in computeStatementsSequence
    statement = statement
  File "/home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/nuitka/optimizations/TraceCollections.py", line 566, in onStatement
    statement.computeStatement(self)
  File "/home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/nuitka/nodes/TryNodes.py", line 128, in computeStatement
    trace_collection = trace_collection
  File "/home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/nuitka/nodes/StatementNodes.py", line 169, in computeStatementsSequence
    statement = statement
  File "/home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/nuitka/optimizations/TraceCollections.py", line 566, in onStatement
    statement.computeStatement(self)
  File "/home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/nuitka/nodes/AssignNodes.py", line 284, in computeStatement
    trace_collection.onExpression(self.getAssignSource())
  File "/home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/nuitka/optimizations/TraceCollections.py", line 535, in onExpression
    trace_collection = self
  File "/home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/nuitka/nodes/ExpressionBases.py", line 820, in computeExpressionRaw
    expression = sub_expression
  File "/home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/nuitka/optimizations/TraceCollections.py", line 535, in onExpression
    trace_collection = self
  File "/home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/nuitka/nodes/ExpressionBases.py", line 820, in computeExpressionRaw
    expression = sub_expression
  File "/home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/nuitka/optimizations/TraceCollections.py", line 535, in onExpression
    trace_collection = self
  File "/home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/nuitka/nodes/ExpressionBases.py", line 820, in computeExpressionRaw
    expression = sub_expression
  File "/home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/nuitka/optimizations/TraceCollections.py", line 535, in onExpression
    trace_collection = self
  File "/home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/nuitka/nodes/ExpressionBases.py", line 841, in computeExpressionRaw
    trace_collection = trace_collection
  File "/home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/nuitka/nodes/CallNodes.py", line 196, in computeExpression
    trace_collection = trace_collection
  File "/home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/nuitka/nodes/ConstantRefNodes.py", line 866, in computeExpressionCall
    call_node    = call_node
  File "/home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/nuitka/optimizations/OptimizeBuiltinCalls.py", line 1409, in computeBuiltinCall
    new_node = _dispatch_dict[builtin_name](call_node)
  File "/home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/nuitka/optimizations/OptimizeBuiltinCalls.py", line 1099, in super_extractor
    builtin_spec  = BuiltinOptimization.builtin_super_spec
  File "/home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/nuitka/optimizations/BuiltinOptimization.py", line 487, in extractBuiltinArgs
    source_ref = node.getSourceReference()
  File "/home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/nuitka/Builtins.py", line 210, in wrapper
    return f(*args, **new_kw)
  File "/home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/nuitka/optimizations/OptimizeBuiltinCalls.py", line 1060, in wrapSuperBuiltin
    if parameter_provider.getParameters().getArgumentCount() > 0:
AttributeError: 'ExpressionOutlineFunction' object has no attribute 'getParameters'
```

**Comment by kayhayen** [comment]
Can you send me the file in question:

> /home/paul/.virtualenvs/nuitka_django/lib/python3.5/site-packages/django/db/models/aggregates.py

That should allow me to reproduce it already. Thanks,
Kay

**Comment by pwhipp** [comment]
[aggregates.py.gz](https://github.com/kayhayen/Nuitka/files/1704723/aggregates.py.gz)

**Comment by kayhayen** [comment]
I managed to reproduce it, thanks.

**Comment by kayhayen** [comment]
The fix will be really simple, I pushed it to factory. Consider http://nuitka.net/doc/factory.html

**Comment by kayhayen** [comment]
Fixed in 0.5.29
